cmd/cmd.go

```diff
@@ -33,3 +51,221 @@ const msgMissingMetadata = `
     Please see https://exercism.io/cli-v1-to-v2 for instructions on how to fix it.
 
 `
+
+type downloadContext struct {
+	usrCfg  *viper.Viper
+	uuid    string
+	slug    string
+	track   string
+	team    string
+	payload *downloadPayload
+}
+
+func newDownload(ctx *downloadContext) error {
+	client, err := api.NewClient(ctx.usrCfg.GetString("token"), ctx.usrCfg.GetString("apibaseurl"))
+	if err != nil {
+		return err
+	}
+
+	req, err := client.NewRequest("GET", ctx.requestURL(), nil)
+	if err != nil {
+		return err
+	}
+
+	query := req.URL.Query()
+	ctx.buildQuery(query)
```

Comment:
Should the `buildQuery` initialize the query itself as well? Something like:

```
query := ctx.buildQuery()
```

Reply:
I don't see a better way of doing it. Initializing requires a reference to the `*http.Request`, right? I don't see how we can avoid having to pass that in.